calls/forms.py

```diff
@@ -1,13 +1,72 @@
+from bsd.models import BSDProfile
 from django import forms
 from django.conf import settings
+from django.contrib.auth.models import User
 from django.forms import widgets
-from .models import CallCampaign
+from django.forms.widgets import Widget
+from django.forms.utils import flatatt
+from django.utils.html import mark_safe
+from .models import CallCampaign, CallProfile
 
 CALLS_MAX_DISTANCE_MILES = settings.CALLS_MAX_DISTANCE_MILES
 CALLS_MAX_LIST_SIZE = settings.CALLS_MAX_LIST_SIZE
 
+class CommaSeparatedCallersTextArea(Widget):
+    def render(self, name, value, attrs=None, renderer=None):
+        final_attrs = self.build_attrs(attrs, {'type':'text', 'name':name})
+        caller_emails = []
+
+        if value is not None:
+            for caller_id in value:
+                callprofile = CallProfile.objects.get(pk=caller_id)
+                user = callprofile.user
+                caller_emails.append(str(user.email))
+
+            value = ', '.join(caller_emails)
+            if value:
+                final_attrs['value'] = str(value)
+        return mark_safe(u'<input%s />' % flatatt(final_attrs))
+
+class ModelCommaSeparatedChoiceField(forms.ModelMultipleChoiceField):
+    widget = CommaSeparatedCallersTextArea(
+        attrs={
+            'rows': '5',
+            'placeholder': 'bob@example.com, sally@example.comm, tom@example.com'
+        }
+    )
+
+    def clean(self, value):
+        caller_ids = []
+
+        if value is not None and value != '':
+            value = [email.strip() for email in value.split(",")]
+
+            for email in value:
+                # TODO: support multiple accounts per email
+                if User.objects.filter(email__iexact=email).exists():
+                    user = User.objects.get(email__iexact=email)
```

Comment:
This is doing 2 different db queries to lookup the same object twice. Please simply this to just one query.

Might also want to use `get_or_create` method: https://docs.djangoproject.com/en/1.11/ref/models/querysets/#get-or-create

Reply:
👍